torch/testing/_internal/common_optimizers.py

```diff
@@ -183,6 +184,32 @@ def test_wrapper(*args, **kwargs):
                 raise ex
 
 
+# ----------------------------------------------------------------------------------------------------------------
+# NOTE: The following optim_inputs_func_* sampling functions only return constructor combinations of NON-IMPLEMENTATION
```

Comment:
Define what these flags are!, in particular I see maximize below which I would expect to be one such flag :D 
Also let's have a test that check that these flags are never provided in the sample inputs to ensure this is followed

Reply:
i have done so!